streams-tests/shared/src/test/scala/zio/stream/ZTransducerSpec.scala

```diff
@@ -74,6 +74,11 @@ object ZTransducerSpec extends ZIOBaseSpec {
         },
         testM("doesn't emit empty trailing chunks") {
           assertM(run(ZTransducer.collectAllN[Int](3), List(Chunk(1, 2, 3))))(equalTo(Chunk(List(1, 2, 3))))
+        },
+        testM("emits chunks when exactly N elements received") {
```

Comment:
The test could be blackboxified: something like
`(ZStream(1,2,3,4) ++ ZStream.fromEffect(ZIO.never)).transduce(t).take(1)`

Reply:
Thanks! I actually prefer to test the transducer directly in this case as it is a bit of an edge case.